src/librustc_mir/interpret/cast.rs

```diff
@@ -223,16 +224,16 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         match dest_ty.kind {
             // float -> uint
             Uint(t) => {
-                let width = t.bit_width().unwrap_or_else(|| self.pointer_size().bits() as usize);
-                let v = f.to_u128(width).value;
+                let width = t.bit_width().unwrap_or_else(|| self.pointer_size().bits());
+                let v = f.to_u128(usize::try_from(width).unwrap()).value;
```

Comment:
This is apfloat using `usize` instead of `u64` for sizes...

Reply:
I mean, it could be an `u8` just as well, the max value here is `128`.

Reply:
IMO it would make sense to consistently use the same type for sizes, that avoids casts. Note how changing `bit_width` to return `u64` got rid of a bunch of casts both in Miri and codegen.